### PR TITLE
jffs2: add NAND flash memory initialization

### PR DESCRIPTION
**Pull request by HBuczynski**
## Motivation and Context

DTR-102

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## How Has This Been Tested?

- [ ] Already covered by automatic testing.
- [ ] New test added: (add PR link here).
- [ ] Tested by hand on: (list targets here).

## Checklist:

- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing linter checks and tests passed.
- [ ] My changes generate no new compilation warnings for any of the targets.

## Special treatment

- [ ] This PR needs additional PRs to work (list the PRs, preferably in merge-order).
- [ ] I will merge this PR by myself when appropriate.

### CODE REVIEW

File: jffs2/phoenix-rtos/mtd.c [hunk 1]
```diff
@@ -39,25 +39,26 @@ struct dentry *mount_mtd(struct file_system_type *fs_type, int flags,
...
-		mtd->size = p->strg->size;
-
-		mtd->index = 0;
-		mtd->oobsize = 0;
-		mtd->oobavail = 0;
-		mtd->storage = p->strg;
 	}
 	else if (p->strg->dev->mtd->type == mtd_nandFlash) {
-		/* TODO */
+		mtd->type = MTD_NANDFLASH;
```
Comment (agkaminski) [inline jffs2/phoenix-rtos/mtd.c:45]: Hmm, I guess this `mtd_nandFlash` is our own const and `MTD_NANDFLASH` is a standard one? If so it would be better to remove `mtd_nandFlash` and use standard one

File: jffs2/libjffs2.c [hunk 2]
```diff
@@ -596,11 +596,6 @@ static int libjffs2_create(void *info, oid_t *dir, const char *name, oid_t *oid,
 				dev_find_oid(p->devs, dev, d_inode(dentry)->i_ino, 1);
 			break;
 		case otSymlink:
-			/* empty target check */
-			if (dentry->d_name.len >= namelen) {
```
Comment (agkaminski) [inline jffs2/libjffs2.c:600]: This change seems somewhat not related. Why is it needed?

**Review by agkaminski — APPROVED** [review]
No merge blockers

File: jffs2/libjffs2.c (hunk 2 above)
Comment (HBuczynski) [reply jffs2/libjffs2.c:600]: https://github.com/phoenix-rtos/phoenix-rtos-filesystems/blob/19cb5240dc4377c7167fb9301ac52192747ecdac/jffs2/libjffs2.c#L562

This condition makes no sense. While NAND has been tested the symlinks couldn't be created.

File: jffs2/phoenix-rtos/mtd.c (hunk 1 above)
Comment (agkaminski) [reply jffs2/phoenix-rtos/mtd.c:45]: In this case `mtd_` prefix is missleading - this enum does not apply to mtd module (libmtd), but to storage. So `storage_` prefix is more appropriate
